### PR TITLE
Entity dummy response and assertions

### PR DESCRIPTION
**Pull request by sadiqkhoja**
- Dummy responses from the Entity API
- Integration tests and assertions

### CODE REVIEW

File: lib/resources/entities.js [hunk 1]
```diff
@@ -0,0 +1,368 @@
+// Copyright 2022 ODK Central Developers
```
Comment (ktuite) [inline lib/resources/entities.js:1]: 2023

File: test/integration/api/entities.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    it('should return extended metadata of the entities of the dataset', testService(async (service) => {
+      const asAlice = await service.login('alice');
+
+      await asAlice.get('/v1/projects/1/datasets/People/entities')
+        .set('X-Extended-Metadata', true)
+        .expect(200)
+        .then(({ body: people }) => {
+          people.forEach(p => {
+            p.should.be.an.Entity();
+            p.should.have.property('currentVersion').which.is.an.EntityDef();
```
Comment (ktuite) [inline test/integration/api/entities.js:42]: Should these be ExtendedEntity and ExtendedEntityDef?

File: test/integration/api/entities.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    // it should reject if uuid is not found
+    // it should reject if uuid in queryParam and body don't match
+    // it should reject if uuid or label is missing
+    // it should reject if property is not present in dataset.publishedProperties
+    // it should reject if user don't have permission
+  });
+
+  describe('PATCH /datasets/:name/entities/:uuid', () => {
+
+    it('should partially update an Entity', testService(async (service) => {
```
Comment (ktuite) [inline test/integration/api/entities.js:193]: I'm trying to understand the intended difference between PUT and PATCH here. Not that we have to figure it out now, though!

**Review by ktuite — APPROVED** [review]
Looks great! I have a much easier time reasoning about this breakdown - thanks for making the changes.

File: test/integration/api/entities.js (hunk 3 above)
Comment (sadiqkhoja) [reply test/integration/api/entities.js:193]: PUT requires complete object. PATCH allows single field to be updated

**Pull request merged**